tests/misc/vhost.conf

```diff
@@ -0,0 +1,44 @@
+#
+# Apache VirtualHost configuration used by Travis CI.
+#
+# This configuration file will be placed into the default Apache configuration
+# on test machines, making it so that the localhost URL will be able to serve
+# pages generated by PHP-FPM.
+#
+<VirtualHost *:80>
+  ServerAdmin webmaster@localhost
+  DocumentRoot "/var/www/web"
+
+  <Directory />
+    Options FollowSymLinks
+    AllowOverride None
+  </Directory>
+
+  # This directory will be replaced with the appropriate web root by the
+  # .travis.yml configuration.
+  <Directory "/var/www/web/">
+    Options Indexes FollowSymLinks MultiViews ExecCGI
+    AllowOverride All
+    Order allow,deny
+    Allow from all
+
+    # Populate $_SERVER['BACKDROP_SETTINGS']. See backdrop_settings_initialize()
+    # and the test in bootstrap.test.
+    #SetEnv BACKDROP_SETTINGS {"databases":{"test_database":{"default":{"driver":"mysql","database":"test_database","username":"test_username","password":"test_password","host":"test_host"}}},"settings":{"test_setting":"foo"}}
+  </Directory>
+
+  # Wire up Apache to use Travis CI's php-fpm.
+  <IfModule mod_fastcgi.c>
+    AddHandler php5-fcgi .php
```

Comment:
Why the oldfashioned mod_fastcgi and why php5 here? You installed php7.2. Or am I missing something?

I'd suggest to use php7.2-fpm + proxy_fcgi_module (the latter also requires fcgid_module).
php-fpm is an extra service running all the php stuff, while apache then only delivers http. That's the modern way of running php. (I'm a bit astounded that this setup seems to work anyway.)

Reply:
Per default (at least in Debian), php-fpm service gets turned on globally, but you need to adapt the user in /etc/php/7.2/fpm/pool.d/php72.conf (default is www-data, but you need travis:travis). But maybe travisCI already does this.

Reply:
I suppose, travis already set up php-fpm correnctly (with php 7.2) the whole block (`<IfModule mod_fastcgi.c>`) could be obsolete.

Another question: Why the old Ubuntu Xenial, when Bionic is available?